core/src/apps/hedera/sign_tx.py

```diff
@@ -0,0 +1,134 @@
+from trezor.crypto.curve import ed25519
+from trezor.messages import HederaSignTx
+from trezor.messages import HederaSignedTx
+from trezor.strings import format_amount
+
+from apps.common import paths
+from apps.common.keychain import auto_keychain
+
+from .layout import (
+    confirm_account,
+    confirm_associate_token,
+    confirm_create_account,
+    confirm_transfer_hbar,
+    confirm_transfer_token,
+    format_account,
+)
+
+transaction_types = {
+    "CONFIRM": "Confirm Account",
+    "CREATE": "Create Account",
+    "TOKEN": "Token Transfer",
+    "HBAR": "Hbar Transfer",
+    "ASSOCIATE": "Token Associate",
+}
+
+
+def validate_tx(msg: HederaSignTx) -> str | None:
```

Comment:
Consider creating more functions that would fully validate submessages of `HederaSignTx`. For example the `cryptoCreateAccount`:
```
def is_valid_crypto_create_tx_body(message: HederaCryptoCreateTransactionBody) -> bool:
    return message.initialBalance is not None

def validate_tx(msg: HederaSignTx) -> str | None:
    if msg.tx is not None:
        # Create Account
        if msg.tx.cryptoCreateAccount is not None:
            # can't be both
            if msg.tx.cryptoTransfer is not None:
                return None

            if is_valid_crypto_create_tx_body(msg.tx.cryptoCreateAccount):
                return transaction_types["CREATE"]

            return None
    ...
```
This example may look dummy, but for other nested messages it can be helpful and once the validation is done, the message can further be safely used.